packages/test/test-end-to-end-tests/src/test/offline/stashedOps.spec.ts

```diff
@@ -1599,6 +1595,52 @@ describeCompat("stashed ops", "NoCompat", (getTestObjectProvider, apis) => {
 		const handle = await timeoutAwait(handleP, {
 			errorMsg: "Timeout on waiting for handleP",
 		});
+		map.set("blob handle", handle);
+
+		const container2 = await timeoutAwait(provider.loadTestContainer(testContainerConfig), {
+			errorMsg: "Timeout on waiting for container2 load",
+		});
+		const dataStore2 = (await container2.getEntryPoint()) as ITestFluidObject;
+		const map2 = await dataStore2.getSharedObject<ISharedMap>(mapId);
+
+		await timeoutAwait(provider.ensureSynchronized(), {
+			errorMsg: "Timeout on waiting for ensureSynchronized after creating container2",
+		});
+
+		const handleGet2: any = await timeoutAwait(map2.get("blob handle").get(), {
+			errorMsg: "Timeout on waiting for handleGet2",
+		});
+		assert.strictEqual(bufferToString(handleGet2, "utf8"), "blob contents");
+	});
+
+	it("blob upload and attach before loading", async function () {
+		const container = await loadContainerOffline(testContainerConfig, provider, { url });
+		const dataStore = (await container.container.getEntryPoint()) as ITestFluidObject;
+		const map = await dataStore.getSharedObject<ISharedMap>(mapId);
+
+		const handle = await dataStore.runtime.uploadBlob(stringToBuffer("blob contents", "utf8"));
+		map.set("blob handle", handle);
```

Comment:
One thing to consider is that in the payloadPending flow, setting the handle in the map will unblock the upload to start, but the upload attempt will not have *actually* started here before you call `container.connect()` (since these are back-to-back sync calls).  Does this test care if the upload starts before calling `container.connect()`?

Reply:
I believe the important part is to make the `this.storage.createBlob` call before container.connect() to make sure we don't have epoch available.